File: java/src/main/java/com/leetcode/OneOneSix.java
package com.leetcode;

/**
 * @author: John
 * @date: 2020-03-19 19:50
 * @description: 116
 **/
public class OneOneSix {
    class Node {
        public int val;
        public Node left;
        public Node right;
        public Node next;

        public Node(int val) {
            this.val = val;
        }

        public Node(int val, Node left, Node right, Node next) {
            this.val = val;
            this.left = left;
            this.right = right;
            this.next = next;
        }
    }

}

File: leetcodecpp/leetcode/OneOneSix.cpp

//
// Created by John on 2020/3/19.
//

#include "Tree.h"

class Solution{

public:
    void connect(PrefectBinaryTree *root) {
        if (root == NULL) {
            return;
        }
        PrefectBinaryTree * pre = root;
        PrefectBinaryTree * cur = NULL;
        while (pre->left) {
            cur = pre;
            while (cur) {
                cur->left->next = cur->right;
                if (cur->next) {
                    cur->right->next = cur->next->left;
                }
                cur = cur->next;
            }
            pre = pre->left;
        }
    }
};
